haystack_experimental/dataclasses/chat_message.py

```diff
@@ -193,3 +196,77 @@ def to_dict(self) -> Dict[str, Any]:
 
         serialized["content"] = content
         return serialized
+    
+    def to_openai_dict_format(self) -> Dict[str, Any]:
+        """
+        Convert a ChatMessage to the dictionary format expected by OpenAI's Chat API.
+        """
+        text_contents = self.texts
+        tool_calls = self.tool_calls
+        tool_call_results = self.tool_call_results
+
+        if not text_contents and not tool_calls and not tool_call_results:
+            raise ValueError(
+                "A `ChatMessage` must contain at least one `TextContent`, `ToolCall`, or `ToolCallResult`."
+            )
+        if len(text_contents) + len(tool_call_results) > 1:
+            raise ValueError("A `ChatMessage` can only contain one `TextContent` or one `ToolCallResult`.")
+
+        openai_msg: Dict[str, Any] = {"role": self._role.value}
+
+        # Add name field if present
+        if self._name is not None:
+            openai_msg["name"] = self._name
+
+        # user message
+        if openai_msg["role"] == "user":
+            if len(self._content) == 1:
+                openai_msg["content"] = self.text
+                return openai_msg
```

Comment:
Just clarifying this works and doesn't need an isinstance check on what `self._content` is because if it's length 1 it has to be text?